code/src/java/pcgen/io/exporttoken/AbilityToken.java

```diff
@@ -254,50 +253,44 @@ protected String getTokenForCategory(String tokenSource,
 			}
 
 			// If it's EXCLUDETYPE and it actually has a value attached then process it 
-			if (extypeInd != -1 && typeStr.length() > 12)
+			if ((extypeInd != -1) && (typeStr.length() > 12))
 			{
 				// exclude TYPEs from comma-separated list
-				StringTokenizer exTok = new StringTokenizer(typeStr.substring(extypeInd + 12), Constants.SEMICOLON);
+				StringTokenizer exTok = new StringTokenizer(
+					typeStr.substring(extypeInd + 12), Constants.SEMICOLON);
 				while (exTok.hasMoreTokens())
 				{
 					negate.add(exTok.nextToken());	
 				}
 			}
-			
+
 			int keyInd = typeStr.indexOf("KEY=");
 			// If it's KEY and it actually has a value attached then process it 
-			if (keyInd != -1 && typeStr.length() > 4)
+			if ((keyInd != -1) && (typeStr.length() > 4))
 			{
 				key = typeStr.substring(keyInd + 4);
 			}
-			
+
 			int aspectInd = typeStr.indexOf("ASPECT=");
 			// If it's ASPECT and it actually has a value attached then process it
-			if (aspectInd != -1 && typeStr.length() > 7)
+			if ((aspectInd != -1) && (typeStr.length() > 7))
 			{
 				aspect = typeStr.substring(aspectInd + 7);
 			}
-			
+
 		}
 
 		// Ability List
 		MapToList<Ability, CNAbility> aList = null;
 		// Build the list of abilities that we should display
-		if (key == null)
-		{
-			aList = AbilityToken.buildAbilityList(types, negate, abilityType,
-					view, aspect, abilityList);
-		}
-		else
-		{
-			aList = AbilityToken.buildAbilityList(key, view, abilityList);
-		}
+		MapToList<Ability, CNAbility> abilityCNAbilityMapToList =
+			(key == null)
```

Comment:
I think this is too complicated for an inline IF.  They should be short, not wrapping lines.